src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/Utilities/SyntacticDocument.cs

```diff
@@ -14,25 +13,23 @@ internal class SyntacticDocument
     {
         public readonly Document Document;
         public readonly SourceText Text;
-        public readonly SyntaxTree SyntaxTree;
         public readonly SyntaxNode Root;
 
-        protected SyntacticDocument(Document document, SourceText text, SyntaxTree tree, SyntaxNode root)
+        protected SyntacticDocument(Document document, SourceText text, SyntaxNode root)
```

Comment:
It might be worth adding the note from the PR description on how this differs from the ParseDocument in the code here as well.

I also wondering - is it possible to replace usages of `SyntacticDocument` with either a regular `Document` or a `ParsedDocument` and get rid of this?

Reply:
More cleanup is certainly possible. I'm trying to limit the size of the PR :)

Reply:
Added to "follow ups" in the description.